### DISCUSSION TITLE
Reolink 520 and recording issues

### DISCUSSION THREAD

**Discussion by TheQue42** [blakeblackshear/frigate#1897] [Ask A Question]
I am a bit confused on how to configure my Reolink 520. 

The docs at  https://docs.frigate.video/configuration/camera_specific#rtmp-cameras-reolink-410520-and-possibly-others, indicate that "The input parameters need to be adjusted for RTMP cameras", but I've got this configured as a normal RTSP camera, just like all my other cams, not RTMP, so what am I missing?

I DO have issues with recording, that fails *intermittently*, with the clip url returning:

"Could not create clip from recordings for <camera>" and the frigate log saying:

```
[2021-10-02 14:40:11] frigate.http                   ERROR   : ffmpeg version 4.3.1 Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-17ubuntu1~20.04)
  configuration: --disable-debug --disable-doc --disable-ffplay --enable-shared --enable-avresample --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-gpl --enable-libfreetype --enable-libvidstab --enable-libmfx --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libxcb --enable-libx265 --enable-libxvid --enable-libx264 --enable-nonfree --enable-openssl --enable-libfdk_aac --enable-postproc --enable-small --enable-version3 --enable-libzmq --extra-libs=-ldl --prefix=/opt/ffmpeg --enable-libopenjpeg --enable-libkvazaar --enable-libaom --extra-libs=-lpthread --enable-vaapi --extra-cflags=-I/opt/ffmpeg/include --extra-ldflags=-L/opt/ffmpeg/lib
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
pipe:: Invalid data found when processing input
```
Then, after a restart everything is all of a sudden, working again, and my config changes may have been things like adding a motion mask???

I wanted to increase ffmpeg logging output, but I dont understand the syntax:

"ffmpeg.<camera_name>.**<sorted_roles>**" and the associated comment "NOTE: All FFmpeg logs are sent as error level."

I'm currently running with:

```
logger:
  default: debug
  logs:
    frigate.mqtt: warning
    peewee.sqliteq: warning
    peewee: warning
    ffmpeg.fr_hallen: debug
```
But that doesnt seem to provide much more ffmpeg debugging.

My release-version installation has been working so flawlessly that I'm either doing something very stupid, or there is something different in ffmpeg logik in the new 0.9 release, but I am having major problems trying to gather troubleshooting info when the problems are so intermittent, and I debug logging output i so sparse.

And I **am** a teacher of troubleshooting, so I am at least not a n00b in that area (compared to ffmpeg stuff)

*Sometimes*, I got better results with the config:
```
      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer
        - -rtsp_transport
        - tcp
        - -flags
        - low_delay
        - -strict
        - experimental
        - -fflags
        - +genpts+discardcorrupt
        - -use_wallclock_as_timestamps
        - "1"
```
But lately I'v run without any custom args. I've then notices the attached artifacts in the frigate UI for the camera.
![reolink-520](https://user-images.githubusercontent.com/38688906/135716966-fe5fef30-2cad-459b-aea0-06a44e2d2c20.jpg)
![reolink-520-b](https://user-images.githubusercontent.com/38688906/135716971-3a68bf6b-3fbc-4b4a-b76c-177e7f596888.jpg)
What should I do to gather more info?

**Comment by blakeblackshear** [comment]
It would be best to open an issue rather than a discussion and provide the info requested in the template. Or provide that info here

Many Reolink users use RTMP to avoid these issues, but if you were using previous versions without rtmp, it should be possible to make it work here too.

**Reply by TheQue42** [reply]
And how do I change to RTMP? My google-fu seems useless, and the Reolink APP and WebGUI has NO mention of RTMP anywhere? :-(

**Reply by TheQue42** [reply]
Sorry, I should have opened a proper ticket of course, but since I'm a bit lost in finding good troubleshooting info, I wanted to get some feedback.
(And a frantic 3year old, and a 3 week old baby is also a bit distracting at the moment)

I'll fill in the proper info in a ticket layer, as soon as I find the time. I have the results of an ffprobe here though, with different results depending on the version.
```
==
RUNNING INSIDE FRIGATE DOCKER CONTAINER
==

root@dc876ce915da:/opt/frigate# ffprobe rtsp://USERNAME:PASSWORD@IP_ADDRESS/h264Preview_01_main
ffprobe version 4.3.1 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-17ubuntu1~20.04)
  configuration: --disable-debug --disable-doc --disable-ffplay --enable-shared --enable-avresample --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-gpl --enable-libfreetype --enable-libvidstab --enable-libmfx --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libxcb --enable-libx265 --enable-libxvid --enable-libx264 --enable-nonfree --enable-openssl --enable-libfdk_aac --enable-postproc --enable-small --enable-version3 --enable-libzmq --extra-libs=-ldl --prefix=/opt/ffmpeg --enable-libopenjpeg --enable-libkvazaar --enable-libaom --extra-libs=-lpthread --enable-vaapi --extra-cflags=-I/opt/ffmpeg/include --extra-ldflags=-L/opt/ffmpeg/lib
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
Input #0, rtsp, from 'rtsp://USERNAME:PASSWORD@10.9.24.206/h264Preview_01_main':
  Metadata:
    title           : Session streamed by "preview"
    comment         : h264Preview_01_main
  Duration: N/A, start: 0.000000, bitrate: N/A
    Stream #0:0: Video: h264, yuv420p(progressive), 2560x1920, 29.97 tbr, 90k tbn, 180k tbc
    Stream #0:1: Audio: aac, 16000 Hz, mono, fltp
root@dc876ce915da:/opt/frigate# exit
16:52-Sat_02
==
RUNNING ON DOCKER HOST, OPENSUSE 15.2
==
#> ffprobe rtsp://USERNAME:PASSWORD@10.9.24.206/h264Preview_01_main
ffprobe version 3.4.4 Copyright (c) 2007-2018 the FFmpeg developers
  built with gcc 7 (SUSE Linux)
  configuration: --prefix=/usr --libdir=/usr/lib64 --shlibdir=/usr/lib64 --incdir=/usr/include/ffmpeg --extra-cflags='-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g' --optflags='-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -g' --disable-htmlpages --enable-pic --disable-stripping --enable-shared --disable-static --enable-gpl --disable-openssl --enable-avresample --enable-libcdio --enable-gnutls --enable-ladspa --disable-cuda --disable-cuvid --enable-libass --enable-libbluray --enable-libcelt --enable-libcdio --enable-libdc1394 --enable-libfreetype --enable-libgsm --enable-libmp3lame --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libzimg --enable-libzvbi --enable-vaapi --enable-vdpau --enable-muxers --enable-demuxers --disable-encoders --disable-decoders --disable-decoder='mpeg4,h263,h264,hevc,vc1' --enable-encoder='apng,ass,ayuv,bmp,ffv1,ffvhuff,flac,gif,huffyuv,jpegls,libgsm,libmp3lame,libopenjpeg,libopus,libschroedinger,libspeex,libtheora,libtwolame,libvorbis,libvpx_vp8,libvpx_vp9,libwebp,libwebp_anim,mjpeg,mp2,mp2fixed,opus,pam,pbm,pcm_alaw,pcm_f32be,pcm_f32le,pcm_f64be,pcm_f64le,pcm_mulaw,pcm_s16be,pcm_s16be_planar,pcm_s16le,pcm_s16le_planar,pcm_s24be,pcm_s24le,pcm_s24le_planar,pcm_s32be,pcm_s32le,pcm_s32le_planar,pcm_s8,pcm_s8_planar,pcm_u16be,pcm_u16le,pcm_u24be,pcm_u24le,pcm_u32be,pcm_u32le,pcm_u8,pcx,pgm,pgmyuv,png,ppm,sgi,srt,ssa,sunrast,targa,text,tiff,v210,v308,v408,v410,vorbis,xbm,xwd,y41p,yuv4,zlib,' --enable-decoder='ac3,ansi,apng,ass,ayuv,bmp,dirac,exr,ffv1,ffvhuff,ffwavesynth,flac,gif,gsm,huffyuv,libcelt,libgsm,libopenjpeg,libopus,libschroedinger,libspeex,libvorbis,libvpx_vp8,libvpx_vp9,mjpeg,mpeg1video,mpeg2video,,mp1,mp1float,mp2,mp2float,mp3,mp3float,opus,pam,pbm,pcm_alaw,pcm_bluray,pcm_dvd,pcm_f32be,pcm_f32le,pcm_f64be,pcm_f64le,pcm_mulaw,pcm_s16be,pcm_s16be_planar,pcm_s16le,pcm_s16le_planar,pcm_s24be,pcm_s24le,pcm_s24le_planar,pcm_s32be,pcm_s32le,pcm_s32le_planar,pcm_s8,pcm_s8_planar,pcm_u16be,pcm_u16le,pcm_u24be,pcm_u24le,pcm_u32be,pcm_u32le,pcm_u8,pcx,pgm,pgmyuv,pgssub,png,ppm,rawvideo,sgi,srt,ssa,sunrast,targa,text,theora,tiff,v210,v210x,v308,v408,v410,vorbis,vp3,vp5,vp6,vp6a,vp6f,vp8,vp9,webp,xbm,xwd,y41p,yuv4,zlib,'
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libavresample   3.  7.  0 /  3.  7.  0
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100
[rtsp @ 0x55834763a000] Could not find codec parameters for stream 0 (Video: h264 (High), none(progressive)): unspecified size
Consider increasing the value for the 'analyzeduration' and 'probesize' options
Input #0, rtsp, from 'rtsp://USERNAME:PASSWORD@10.9.24.206/h264Preview_01_main':
  Metadata:
    title           : Session streamed by "preview"
    comment         : h264Preview_01_main
  Duration: N/A, start: 0.000000, bitrate: N/A
    Stream #0:0: Video: h264 (High), none(progressive), 90k tbr, 90k tbn, 90k tbc
    Stream #0:1: Audio: aac, 16000 Hz, 1 channels
Unsupported codec with id 28 for input stream 0
Unsupported codec with id 86018 for input stream 1

```

**Reply by TheQue42** [reply]
Tried RC6 with similar results (although using http instead of rtsp). There at least seems to a bit of lack in the detection of the recording results, since I now have several events in the frigate UI, listing person detections on the fr_hallen camera and with (what looks like) clickable movie-clips.
The same clips are also available to see (at least it looks like it via the browser) via HA's media browser.

But in all cases the movie cant be played, and I cant find any mp4's matching that timeslot in the media-dir. But I can find the same error logs as above, for those times in the frigate container log.
So it seems that frigate doesnt really detect, that the recordings arent saved/created?

I will try to write a real TR as soon as I have the time, but could you explain as I asked before, about the:

"ffmpeg.<camera_name>.<sorted_roles>" debug syntax so I can see if I can get more debug output?

**Reply by blakeblackshear** [reply]
Any log messages with the ffmpeg prefix are just a direct copy of what ffmpeg outputs. When ffmpeg exits, frigate restarts the process and outputs the last 100 lines of output from the ffmpeg process. All messages from ffmpeg are type ERROR regardless of what the type was from ffmpeg.

**Comment by jasonpstokes** [comment]
Reolink's RTMP support page is here: https://support.reolink.com/hc/en-us/articles/360038791973-Is-Reolink-Camera-Compatible-with-Streaming-on-YouTube-Live if you want to try that.

However, I've found the http:// streams to be the most stable for my Reolink 520A cameras (over RTMP and RTSP) - my old RC2 docker setup and Frigate config are in this post [here](https://github.com/blakeblackshear/frigate/issues/1567#issuecomment-911557560) if you want to try that as well.

**Reply by TheQue42** [reply]
I've changed the path-url to http, but with no changes in the record-issue at least. And it still seems like the problem is intermittent, or only occurs in the beginning of a frigate start, and leaving recording on for a long while, with lots of attempts that fail, it will eventually start working.

**Comment by trythatagain** [comment]
This is what I've used in Shinobi, when I was having some issues, mine is a RLC 420 though, but it might be worth a try.

rtmp://username:password@192.168.1.11/bcs/channel0_main.bcs?token=sdasdasd&channel=0&stream=0&user=username&password=password
